interpreter.asm

```diff
@@ -741,3 +723,55 @@ OLD_BASE = * + 1
 
 .chars_to_process
     !byte 0
+
++BACKLINK "unused", 6
+UNUSED
+    jsr LATEST
+    jsr HERE
+    jmp MINUS
+
++BACKLINK "dowords", 7 ; ( xt -- )
+    ; per the standard, nothing must be left on stack before execute
```

Comment:
the comment is outdated - because it's not standard - but i think the behavior is valuable and it would be good to mention in docs, if not already

Reply:
I just happened to remember, I already implemented `TRAVERSE-WORDLIST` a while ago, but reverted it since I didn't have any real use for it:

```
    +BACKLINK
    !byte 17
    !text "traverse-wordlist"
TRAVERSE_WORDLIST ; ( xt -- )
    lda LSB,x
    sta .xt
    lda MSB,x
    sta .xt + 1
    inx
    jsr LATEST
-   jsr FETCH
    lda MSB,x
    beq .ret
    pha
    lda LSB,x
    pha
.xt = * + 1
    jsr PLACEHOLDER_ADDRESS
    lda MSB,x
    ora LSB,x
    bne +
    pla
    pla
    jmp .ret
+   pla
    sta LSB,x
    pla
    sta MSB,x
    jmp -
.ret
    inx
    rts
```

Reply:
It's not only required, it's necessary.  XT must be able to rely on no distance between NT and the rest of the stack of it's going to use it for reference or return values.

As to that code, I'll adapt the self modifying bit in but as headers are no longer in a linked list, advancing the dictionary will be the bulk of the code.